include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -257,8 +252,18 @@ struct is_sycl_iterator<oneapi::dpl::__internal::sycl_iterator<Mode, Types...>>
 {
 };
 
+template <typename Iter, typename Void = void>
```

Comment:
I guess right now we can remove `is_hetero_legacy_trait` stuff because we have 
`struct is_passed_directly<Iter, ::std::enable_if_t<Iter::is_passed_directly::value>>` specialization defined above.

Reply:
These are two separate traits, `is_hetero` and `is_passed_directly`.  Am I missing something?

Reply:
yes, separate traits.
Sorry, I missed that.